docs/new_flux_ideas_and_guide.md

```diff
@@ -0,0 +1,985 @@
+| --- | --- | --- | --- | --- |
+| coyote_creek | CA | 48 | 50 | 48.714 |
+| puget_sound | CA | 40 | 40 | 40 |
+| santa_monica | WA | 50 | 51 | 50.5 |
+
+Although we could have left out the `state` grouping from our example because in this case the `location` is 
+enough to unique identify a series.
+
+Say we wanted to get a summary per state: 
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees, time > -3h}
+    |> aggregate({min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)}, by: ["state"]) 
+```
+
+| state | min_bottom_degrees | max_bottom_degrees | mean_bottom_degrees |
+| --- | --- | --- | --- |
+| CA | 48 | 51 | 49.11 |
+| WA | 40 | 40 | 40 |
+
+See that the results don't have the `location` column even though we selected it on the first line. That's because 
+it wasn't included in the `by` set of columns in the call to `aggregate`.
+
+The aggregate function can window the data by time intervals. For example, if we wanted to summarize each hour for 
+the last three hours:
+
+```
+@my_data.h2o_temperature{location, bottom_degrees, time > -3h}
+    |> aggregate({min(bottom_degrees), max(bottom_degrees), mean(bottom_degrees)}, by: ["location"], window: 1h) 
+```
+
+Because the `aggregate` function is breaking the records out into one hour intervals, the `time` column is kept, 
+but the values are updated to the end of the window. For example, data at 2020-02-22T16:05 would get summarized in 
+a record with the `time` value (window) of `2020-02-22T17:00`. 
+
+| location | min_bottom_degrees | max_bottom_degrees | mean_bottom_degrees | time |
+| --- | --- | --- | --- | --- |
+| santa_monica | 48 | 50 | 48.714 | 2020-02-22T16:00 |
+| santa_monica | 48 | 50 | 48.714 | 2020-02-22T17:00 |
+| santa_monica | 48 | 50 | 48.714 | 2020-02-22T18:00 |
+| coyote_creek | 51 | 51 | 50 | 2020-02-22T17:00 |
+| coyote_creek | 50 | 50 | 50 | 2020-02-22T18:00 |
+| puget_sound | 40 | 40 | 40 | 2020-02-22T18:00 |
+
+Remember there were three series, so for 1h summaries for the last 3 hours we'd expect to have three rows per group 
+for nine total rows. But looking at the results, we only see the full three summaries for the `santa_monica` location. 
+This is because the other two series don't have data in every window of time.
+
+You can create rows for the missing summaries by calling `interpolate` after the call to aggregate. It can be used to 
+fill in default values or values based on a summary statistic of the rows like `first`, `min`, `max`, `mean`, etc. You 
+can find the [details for interpolate in the function reference]().
+
+#### Select
+The `select` function works similar to `aggregate` in that it operates on windows of time and groupings of rows.
+However, The difference is that select returns specific rows, rather than a summary data point. Here are a few 
+examples:
+
+```
+@my_data.h2o_temperature{location, state, bottom_degrees, surface_degrees, time > -3h}
```

Comment:
I find this select misleading. This immediately reminds me of SQL but in reverse, and starts to play games with my limited brain capacity… Any way it could be renamed to something else that isn’t reminiscent of SQL?

Reply:
Do you think this would the kind of thing you get over quickly?  Or do you have a suggestion for a better name? 

I share the concern but not sure what to do about it.

Reply:
hmm, I'm not worried about me per say, I'm worried about the person who is coming into flux as a newb.  I wonder if @sanderson could help us out here?

select feels so ingrained and has so much baggage from its use in SQL, my biggest concern here.